var/spack/repos/builtin/packages/py-pylint/package.py

```diff
@@ -31,6 +37,7 @@ class PyPylint(PythonPackage):
     depends_on('py-astroid@2.0:', type=('build', 'run'), when='@2.2.0:')
     depends_on('py-astroid@2.2.0:2', type=('build', 'run'), when='@2.3.0:2.7')
     depends_on('py-astroid@2.5.6:2.6', type=('build', 'run'), when='@2.8.0:')
```

Comment:
```suggestion
    depends_on('py-astroid@2.5.6:2.6', type=('build', 'run'), when='@2.8.0:2.10')
```
Otherwise this constraint will also apply to the latest version.